File: tm/NativeSampleModule.ts
import type { TurboModule } from 'react-native/Libraries/TurboModule/RCTExport';
// import type {TurboModule} from 'react-native'; in future versions
import { TurboModuleRegistry } from 'react-native';

export interface Spec extends TurboModule {
  readonly reverseString: (input: string) => string;
  readonly stringLength: (input: string) => number;
}

export default TurboModuleRegistry.getEnforcing<Spec>('NativeSampleModule');

File: tm/NativeSampleModule.cpp
#include "NativeSampleModule.h"

namespace facebook::react {

NativeSampleModule::NativeSampleModule(std::shared_ptr<CallInvoker> jsInvoker)
    : NativeSampleModuleCxxSpec(std::move(jsInvoker)) {}

std::string NativeSampleModule::reverseString(jsi::Runtime& rt, std::string input) {
  return std::string(input.rbegin(), input.rend());
}

size_t NativeSampleModule::stringLength(jsi::Runtime& rt, std::string input) {
  return input.length(); // Implementation of getStringLength method.
}

} // namespace facebook::react